pkg/api/types.go

```diff
@@ -227,7 +227,7 @@ type WindowsProfile struct {
 	WindowsSku                string            `json:"windowsSku"`
 	WindowsDockerVersion      string            `json:"windowsDockerVersion"`
 	Secrets                   []KeyVaultSecrets `json:"secrets,omitempty"`
-	SSHEnabled                bool              `json:"sshEnabled,omitempty"`
+	SSHEnabled                *bool             `json:"sshEnabled,omitempty"`
```

Comment:
Do we know if we have data in AKS RP with this value? Will a "false" value in DB cause deserialization error to *bool?

Reply:
By default, it is set to true for Windows in AKS.

Reply:
I also used below code to test it.
```
package main
import (
    "encoding/json"
    "fmt"
)

type SomeStruct struct {
    SomeValue *bool `json:"some_value,omitempty"`
}

type OtherStruct struct {
    SomeValue bool `json:"some_value,omitempty"`
}

func main() {
    t := new(bool)
    f := new(bool)

    *t = true
    *f = false

    s1, _ := json.Marshal(SomeStruct{nil})
    s2, _ := json.Marshal(SomeStruct{SomeValue: t, })
    s3, _ := json.Marshal(SomeStruct{SomeValue: f, })

    fmt.Println(string(s1))
    fmt.Println(string(s2))
    fmt.Println(string(s3))

    o1 := OtherStruct{}
    if err := json.Unmarshal(([]byte)(s1), &o1); err != nil {
        fmt.Printf("Error is %s\n", err)
    }
    fmt.Printf("o1: %+v\n", o1)

    o2 := OtherStruct{}
    if err := json.Unmarshal(([]byte)(s2), &o2); err != nil {
        fmt.Printf("Error is %s\n", err)
    }
    fmt.Printf("o2: %+v\n", o2)

    o3 := OtherStruct{}
    if err := json.Unmarshal(([]byte)(s3), &o3); err != nil {
        fmt.Printf("Error is %s\n", err)
    }
    fmt.Printf("o1: %+v\n", o3)
}
```

Reply:
I used below script to test it and it should work
```
package main
import (
    "encoding/json"
    "fmt"
)

type SetStruct struct {
    SomeValue bool `json:"some_value,omitempty"`
}

func (s *SetStruct) String() string {
    return fmt.Sprintf("SetStruct{ SomeValue: %v }", s.SomeValue)
}

type GetStruct struct {
    SomeValue *bool `json:"some_value,omitempty"`
}

func (s *GetStruct) String() string {
    if s.SomeValue == nil {
        return "GetStruct{ SomeValue: nil }"
    } else {
        return fmt.Sprintf("GetStruct{ SomeValue: *%v }", *s.SomeValue)
    }
}

func MarshalStruct(object interface{}) []byte {
    jsonBytes, err := json.Marshal(object)
    if err != nil {
        return nil
    }
    return jsonBytes
}

func UnmarshalStruct(data []byte)*GetStruct  {
    object := GetStruct{}
    err := json.Unmarshal(data, &object)
    if err != nil {
        return nil
    }
    return &object 
}

func main() {

    o1 := SetStruct{SomeValue: true, }
    o2 := SetStruct{SomeValue: false, }

    data1 := MarshalStruct(o1)
    fmt.Println(string(data1 ))

    o11 := GetStruct{}
    if err := json.Unmarshal(data1, &o11); err != nil {
        fmt.Printf("Error is %s\n", err)
    }
    fmt.Printf("o1: %v\n", &o11)

    data2 := MarshalStruct(o2)
    fmt.Println(string(data2 ))
    o22 := GetStruct{}
    if err := json.Unmarshal(data2, &o22); err != nil {
        fmt.Printf("Error is %s\n", err)
    }
    fmt.Printf("o2: %v\n", &o22)
}
```
Output:
```
{"some_value":true}
o1: GetStruct{ SomeValue: *true }
{}
o2: GetStruct{ SomeValue: nil }
```